docs/source/reference/concept.md

```diff
@@ -0,0 +1,43 @@
+# Concept and FAQs
+
+There are a few terms used in `mosec`.
+
+- `worker`: a Python process that executes the `forward` method (inherit from [`mosec.Worker`](mosec.worker.Worker))
+- `stage`: one processing unit in the pipeline, each stage contains several `worker` replicas
+  - each stage retrieves the data from the previous stage and passes the result to the next stage
+  - retrieved data will be deserialized by the [`Worker.deserialize_ipc`](mosec.worker.Worker.deserialize_ipc) method
+  - passed data will be serialized by the [`Worker.serialize_ipc`](mosec.worker.Worker.serialize_ipc) method
+- `ingress/egress`: the first/last stage in the pipeline
+  - ingress gets data from the client, while egress sends data to the client
+  - data will be deserialized by the ingress [`Worker.serialize`](mosec.worker.Worker.serialize) method and serialized by the egress [`Worker.deserialize`](mosec.worker.Worker.deserialize) method
+- `pipeline`: a chain of processing stages
+- `dynamic batching`: batch requests until either the max batch size or the max wait time is reached
+- `controller`: a Rust tokio thread that works on:
```

Comment:
Do we still have this concept?

Reply:
There is no `controller` in the code. But there is a `communicate` function that does something like that. I don't know how to name it.